src/java.base/share/classes/java/lang/runtime/ObjectMethods.java

```diff
@@ -179,8 +179,12 @@ private static int hashCombiner(int x, int y) {
     private static boolean eq(char a, char b) { return a == b; }
     private static boolean eq(int a, int b) { return a == b; }
     private static boolean eq(long a, long b) { return a == b; }
-    private static boolean eq(float a, float b) { return Float.compare(a, b) == 0; }
-    private static boolean eq(double a, double b) { return Double.compare(a, b) == 0; }
+    private static boolean eq(float a, float b) {
+        return Float.floatToRawIntBits(a) == Float.floatToRawIntBits(b);
```

Comment:
`ObjectMethods` are used for records as well and so these methods should not be modified.  Instead I suggest to add new methods in `ValueObjectMethods` class for value object substutitability comparison of float/double.

Reply:
I'm beginning to doubt the wisdom of a difference in field comparisons between value objects and records (and the `==` operator).

Reply:
I believe the discrepancy in `equals` and `==` is valid; value objects may contain reference pointers too, thus they still should be compared by `equals` instead of `==`. Having double fields act like reference fields is fine.